src/AppInstallerCommonCore/Public/winget/Registry.h

```diff
@@ -259,10 +265,23 @@ namespace AppInstaller::Registry
         static Key OpenIfExists(HKEY key, std::string_view subKey = {}, DWORD options = 0, REGSAM access = KEY_READ);
         static Key OpenIfExists(HKEY key, const std::wstring& subKey = {}, DWORD options = 0, REGSAM access = KEY_READ);
 
+        // Creates a new Key or returns one if it already existed. 
+        static Key CreateKeyAndOpen(HKEY key, std::string_view subkey = {}, DWORD options = REG_OPTION_NON_VOLATILE, REGSAM access = KEY_ALL_ACCESS);
+        static Key CreateKeyAndOpen(HKEY key, const std::wstring& subKey = {}, DWORD options = REG_OPTION_NON_VOLATILE, REGSAM access = KEY_ALL_ACCESS);
+
+        // Delete a key
+        static bool DeleteKey(HKEY key, std::string_view subkey = {}, DWORD samDesired = KEY_WOW64_64KEY);

```

Comment:
and this `Delete`, no need to keep mentioning "key" when you need to reference the type, a la `Key::DeleteKey`